surveys/2024-annual-survey/questions.md

```diff
@@ -333,33 +333,39 @@ Type: select one (optional)
 > 
 > We want to find out if people prefer stability and fewer changes, or if they want to see more features being stabilized.  
 
-### Which unimplemented or (nightly only) features are you looking for to be stabilized?
+### Which features are you looking for to be implemented or stabilized?
 
-Please mention here features for the *Rust compiler* that are currently behind a feature gate (only available using an unstable/nightly release of the Rust compiler) or a feature that is missing and in your opinion would be beneficial to the Rust ecosystem or to your work. This list excludes other tooling around the compiler such as cargo, rustup, rustfmt, etc.
+Please mention *Rust compiler or standard library* features that are currently unstable (only available using a nightly release of the Rust compiler) or missing that would in your opinion be beneficial to the Rust ecosystem or to you. This list excludes other tooling around the compiler such as cargo, rustup, rustfmt, etc.
 
 Type: matrix (optional)
 
 Features:
 
 - Specialization ([RFC#1210](https://github.com/rust-lang/rust/issues/31844))
 - Generators/coroutines ([RFC#2033](https://github.com/rust-lang/rust/issues/43122))
-- Async closures ([RFC#2532](https://github.com/rust-lang/rust/issues/62290))
+- Async generators/coroutines (https://github.com/rust-lang/rust/pull/118420)
 - If/while let chains ([eRFC#2497](https://github.com/rust-lang/rust/issues/53667))
 - Try blocks ([RFC#243](https://github.com/rust-lang/rust/issues/31436))
 - Never type ([RFC#1216](https://github.com/rust-lang/rust/issues/35121))
-- Improved traits (trait alias, implied bounds, associated type defaults)
-- Improved const (generic const expressions, const expr for custom types, const trait methods)
-- Compile time reflection (variadic generics)
+- Trait aliases
+- Type alias impl trait (TAIT)
+- Associated type defaults
+- Generic const expressions
+- Const trait methods
+- Compile time reflection
+- Variadic generics
+- Arbitrary self types
 - Enum variant types ([RFC#2593](https://github.com/rust-lang/lang-team/issues/122))
-- Allocator trait, better OOM handling (https://github.com/rust-lang/rust/issues/32838)
+- Allocator trait and better OOM handling (https://github.com/rust-lang/rust/issues/32838)
 - Stable ABI (https://github.com/rust-lang/rust/issues/111423)
+- Portable SIMD (https://github.com/rust-lang/portable-simd)
 
 Priority:
 
-- High Priority
-- Medium Priority
-- Low Priority
-- Should not be prioritised
+- Need this
+- Nice to have
+- Don't need this
+- Don't know what it is
```

Comment:
oh these are great, nice

Reply:
although actually no, I don't like this. I would focus more on "this solves a blocker I have", "this solves a problem I have", "this does not solve a problem I have" (can be reworded a bit better). people often *need* a specific feature, but a specific feature might resolve a blocker they have (that could maybe also be resolved differently)

Reply:
The new "priority" options are sound better than last year, but I'd prefer them to be worded in a way to avoid that slight bitter taste when reading them (see [my comment](https://github.com/rust-lang/surveys/pull/278#discussion_r1749909270) in a related topic).

I like the wording of Nora.

Reply:
I was quite clueless on designing these answers, I just knew that the priority answers were bad :laughing: So I'm open to any suggestions. What about something like this?

```
- Would unblock my use-case
- Would improve my code
- Don't need it
- Don't know what it is
```

Btw, I'm not sure if we should even keep this question (see discussion in https://rust-lang.zulipchat.com/#narrow/stream/219381-t-libs/topic/Rust.20survey.202024). Maybe it would be really better to focus on the problems that users have, rather than on the features.